azext_iot/tests/central/test_iot_central_int.py

```diff
@@ -472,6 +479,80 @@ def test_central_device_registration_summary(self):
         assert json_result[DeviceStatus.blocked.value] is not None
         assert len(json_result) == 4
 
+    def test_central_device_should_start_failover_and_failback(self):
+
+        # created device template & device
+        (template_id, _) = self._create_device_template()
+        (device_id, _) = self._create_device(instance_of=template_id, simulated=False)
+
+        credentials = self.cmd(
+            "iot central device show-credentials --device-id {} --app-id {} --token \"{}\" --central-dns-suffix \"{}\" ".format(
+                device_id, APP_ID, TOKEN, DNS_SUFFIX
+            ),
+        ).get_output_in_json()
+
+        # connect & disconnect device & wait to be provisioned
+        self._connect_gettwin_disconnect_wait_tobeprovisioned(device_id, credentials)
+
+        # initiating failover
+        result = self.cmd(
+            "iot central device manual-failover"
+            " --app-id {}"
+            " --device-id {}"
+            " --ttl {}"
+            " --token \"{}\""
+            " --central-dns-suffix \"{}\""
+            .format(APP_ID, device_id, 5, TOKEN, DNS_SUFFIX)
+        )
+        json_result = result.get_output_in_json()
+
+        # check if failover started and getting original hub identifier
+        hubIdentifierOriginal = json_result["hubIdentifier"]
+        assert len(hubIdentifierOriginal) > 0
+
+        # connect & disconnect device & wait to be provisioned
+        self._connect_gettwin_disconnect_wait_tobeprovisioned(device_id, credentials)
+
+        #Initiating failback
+        fb_result = self.cmd(
+            "iot central device manual-failback"
+            " --app-id {}"
+            " --device-id {}"
+            " --token \"{}\""
+            " --central-dns-suffix \"{}\""
+            .format(APP_ID, device_id, TOKEN, DNS_SUFFIX)
+        )
+
+        # checking if failover has been done by comparing original hub identifier with hub identifier after failover is done
+        fb_json_result = fb_result.get_output_in_json()
+        hubIdentifierFailOver = fb_json_result["hubIdentifier"]
+        assert len(hubIdentifierFailOver) > 0
```

Comment:
please leave all assertions to the very end of the test, otherwise if there is a failure we won't do any of the cleanup ops